### ISSUE TITLE
Dynamic template/partial per item in array

### ISSUE THREAD

**Issue by ghost** [ractivejs/ractive#1096]
# TL;DR
- Is there a way to do dynamic partials which are different on each item in an array?
- Is there a way to re-render an already rendered component?
# Situation

I have tree node data that is contained in an array, which are to be rendered using templates depending on their `type` property. Dynamic partials didn't seem to work this way because I don't see any option to get the data in the context of the partial's position.

```
{{#nodes}}
  {{>dynaPartial}}
{{/nodes}}
```

```
partials : {
  dynaPartial : function(data,helpers){
    // How do I get nodes[i]?
  }
},
```

This would have been the most straightforward approach, if only `dynaPartial` was passed the data in that iteration of `nodes` so I can check for `type`.

Anyways, I have searched for an alternative, and found this approach: It uses a dummy component and using `beforeInit`, set its template accordingly:

```
{{#comps}}
  <compo type="{{type}}" data="{{data}}" />
{{/comps}}
```

```
var Compo = Ractive.extend({
  template : '{{>noode}}',
  beforeInit : function(options){
    var type = options.data.type;
    var template = document.getElementById(type).innerHTML;
    options.partials.noode = template;
  }
});

var inst = new Ractive({
  el : document.body,
  append : true,
  template : '#temp',
  data : {
    comps : [{
      data : 'a',
      type : 'bold'
    }]
  },
  components : {
    compo : Compo
  }
});
```

http://jsbin.com/masayafo/1/edit?html,js,output (click the buttons left to right)

This worked fine until I had to change the `type` of an already rendered node, which doesn't seem to work. Already considered splicing the old and pushing in the updated value, but that seems too hacky. Dummy components are also very hacky :(
